azext_iot/deviceupdate/commands_account.py

```diff
@@ -0,0 +1,223 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+from knack.log import get_logger
+from azext_iot.common.utility import handle_service_exception
+from azext_iot.deviceupdate.common import (
+    ADUPublicNetworkAccessType,
+    ADUAccountSKUType
+)
+from azext_iot.deviceupdate.providers.base import (
+    DeviceUpdateMgmtModels,
+    DeviceUpdateAccountManager,
+    parse_account_rg,
+    ARMPolling,
+)
+
+logger = get_logger(__name__)
+
+# Accounts
+
+
+def create_account(
+    cmd,
+    name,
+    resource_group_name,
+    location=None,
+    tags=None,
+    public_network_access=ADUPublicNetworkAccessType.ENABLED.value,
+    assign_identity=None,
+    scopes=None,
+    role="Contributor",
+    sku=ADUAccountSKUType.STANDARD.value,
+):
+    account_manager = DeviceUpdateAccountManager(cmd=cmd)
+    if not location:
+        location = account_manager.get_rg_location(
+            resource_group_name=resource_group_name
+        )
+
+    account_manager = DeviceUpdateAccountManager(cmd=cmd)
+    identity = account_manager.assemble_account_auth(assign_identity)
+    account = DeviceUpdateMgmtModels.Account(
+        location=location,
+        tags=tags,
+        public_network_access=public_network_access,
+        identity=identity,
+        sku=sku
+    )
+
+    def rbac_handler(lro: ARMPolling):
+        if not scopes:
+            return
+
+        instance = lro.resource().as_dict()
+        identity = instance.get("identity", {})
+        if identity:
+            identity_type: str = identity.get("type")
+            principal_id: str = identity.get("principal_id")
+
+            if (
+                principal_id
+                and scopes
+                and identity_type
+                and "SystemAssigned" in identity_type
+            ):
+                for scope in scopes:
+                    logger.info(
+                        "Applying rbac assignment: Principal Id: {}, Scope: {}, Role: {}".format(
+                            principal_id, scope, role
+                        )
+                    )
+                    logger.info(
+                        account_manager.assign_msi_scope(
+                            scope=scope, principal_id=principal_id, role=role
+                        )
+                    )
+
+    try:
+        create_poller = account_manager.mgmt_client.accounts.begin_create(
+            resource_group_name=resource_group_name, account_name=name, account=account
+        )
+        create_poller.add_done_callback(rbac_handler)
+        return create_poller
+    except Exception as e:
+        handle_service_exception(e)
+
+
+def update_account(cmd, parameters: DeviceUpdateMgmtModels.Account):
```

Comment:
this works?

Reply:
Yes. In fact there are tests that exercise it.  This style is the minimum bar for mvp which supports CLI core generic update params. In a future iteration, there can be specific property update params/args to reduce complexity in using generic update args.